### PR TITLE
executor: fix analyze prefix clustered index get wrong histogram buckets

### LINKED ISSUE #23193: analyze prefix clustered index get wrong histogram buckets
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
drop table if exists t0;
drop table if exists t1;
set @@tidb_enable_clustered_index = 0;
create table t0(a int, b varchar(20), primary key(b(3)));
set @@tidb_enable_clustered_index = 1;
create table t1(a int, b varchar(20), primary key(b(3)));
insert into t0 values (1, '11111'),(2,'222222');
insert into t1 values (1, '11111'),(2,'222222');
analyze table t0;
analyze table t1;
show stats_buckets;
```

### 2. What did you expect to see? (Required)
buckets of t0 and t1 are the same.

### 3. What did you see instead (Required)
```sql
+---------+------------+----------------+-------------+----------+-----------+-------+---------+----------------------+----------------------+------+
| Db_name | Table_name | Partition_name | Column_name | Is_index | Bucket_id | Count | Repeats | Lower_Bound          | Upper_Bound          | Ndv  |
+---------+------------+----------------+-------------+----------+-----------+-------+---------+----------------------+----------------------+------+
| test    | t1         |                | PRIMARY     |        1 |         0 |     1 |       1 | 11111                | 11111                |    0 |
| test    | t1         |                | PRIMARY     |        1 |         1 |     2 |       1 | 222222               | 222222               |    0 |
| test    | t0         |                | PRIMARY     |        1 |         0 |     1 |       1 | 111                  | 111                  |    0 |
| test    | t0         |                | PRIMARY     |        1 |         1 |     2 |       1 | 222                  | 222                  |    0 |
+---------+------------+----------------+-------------+----------+-----------+-------+---------+----------------------+----------------------+------+
```

### 4. What is your TiDB version? (Required)

nightly

### PR DESCRIPTION
**Pull request by lzmhhh123**
Signed-off-by: lzmhhh123 <lzmhhh123@gmail.com>

### What problem does this PR solve?

Issue Number: close #23193

### What is changed and how it works?

How it Works: do not analyze mixed when cluster index has prefix column.

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU

### Release note 

- none

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -1662,7 +1662,7 @@ func (b *PlanBuilder) buildAnalyzeTable(as *ast.AnalyzeTableStmt, opts map[ast.A
 			}
 		}
 		for _, idx := range idxInfo {
-			if idx.Primary && tbl.TableInfo.IsCommonHandle {
+			if idx.Primary && tbl.TableInfo.IsCommonHandle && !idx.HasPrefixIndex() {
```
Comment (coocood) [inline planner/core/planbuilder.go:1665]: Need comment about why doesn't support prefix index.

Comment (lzmhhh123) [reply planner/core/planbuilder.go:1667]: Done

**Comment by coocood** [comment]
```
FAIL: analyze_test.go:393: testFastAnalyze.TestFastAnalyze
[2021-03-10T03:59:55.637Z] 
[2021-03-10T03:59:55.637Z] analyze_test.go:517:
[2021-03-10T03:59:55.637Z]     c.Assert(err.Error(), Equals, "Fast analyze hasn't reached General Availability and only support analyze version 1 currently.")
[2021-03-10T03:59:55.637Z] ... obtained string = "[stats]: global statistics for partitioned tables only available in statistics version2, please set tidb_analyze_version to 2"
[2021-03-10T03:59:55.637Z] ... expected string = "Fast analyze hasn't reached General Availability and only support analyze version 1 currently."
```

File: executor/analyze_test.go [hunk 2]
```diff
@@ -514,7 +514,7 @@ func (s *testFastAnalyze) TestFastAnalyze(c *C) {
 	tk.MustExec("create table t4(a int, b int) PARTITION BY HASH(a) PARTITIONS 2;")
 	tk.MustExec("insert into t4 values(1,1),(3,3),(4,4),(2,2),(5,5);")
 	err = tk.ExecToErr("analyze table t4;")
-	c.Assert(err.Error(), Equals, "Fast analyze hasn't reached General Availability and only support analyze version 1 currently.")
+	c.Assert(err.Error(), Equals, "[stats]: global statistics for partitioned tables only available in statistics version2, please set tidb_analyze_version to 2")
```
Comment (coocood) [inline executor/analyze_test.go:517]: but the session already executed 
`tk.MustExec("set @@tidb_analyze_version = 2;")`

Comment (lzmhhh123) [reply executor/analyze_test.go:517]: Emmmm...It seems something wrong. @Reminiscent @winoros PTAL here.

**Pull request merged**